src/Compilers/Core/Portable/SourceFileResolver.cs

```diff
@@ -135,8 +135,10 @@ public override bool Equals(object? obj)
             return Equals((SourceFileResolver)obj);
         }
 
-        public bool Equals(SourceFileResolver other)
+        public bool Equals(SourceFileResolver? other)
         {
+            RoslynDebug.Assert(other is object);
```

Comment:
>RoslynDebug.Assert(other is object) [](start = 12, length = 35)

This is a `public` method so we should handle `null`.

Reply:
Thanks, indeed. I'll add a null check (since implementation below would crash on dereference...)

---
In reply to: [377720266](https://github.com/dotnet/roslyn/pull/41336#discussion_r377720266) [](ancestors = 377720266)